xds/src/test/java/io/grpc/xds/ClientXdsClientTestBase.java

```diff
@@ -1488,14 +1494,48 @@ protected abstract Message buildLbEndpoint(String address, int port, String heal
     protected abstract Message buildDropOverload(String category, int dropPerMillion);
   }
 
-  static Listener buildListenerWithFilterChain(
+  @SuppressWarnings("deprecation")
+  static FilterChain buildFilterChain(FilterChainMatch filterChainMatch,
```

Comment:
Can you put these methods for building xDS messages into [`MessageFactory`](https://github.com/grpc/grpc-java/blob/f872c6059dfe3c28400db7c1e760a95385aecc14/xds/src/test/java/io/grpc/xds/ClientXdsClientTestBase.java#L1425) for v3? For v2, you can just throw UnSupportedOperation exception. See how [HttpFilter](https://github.com/grpc/grpc-java/blob/f872c6059dfe3c28400db7c1e760a95385aecc14/xds/src/test/java/io/grpc/xds/ClientXdsClientV2Test.java#L280) tests does this.

Reply:
Done